src/fairness/print_hierarchy/Makefile.am

```diff
@@ -8,16 +8,21 @@ AM_CPPFLAGS = $(FLUX_CORE_CFLAGS)
 
 SUBDIRS = .
 
-noinst_PROGRAMS = print_hierarchy
+noinst_PROGRAMS = flux-shares
 
-print_hierarchy_SOURCES = \
-    print_hierarchy.cpp
+flux_shares_SOURCES = flux_shares.cpp \
+                      ../reader/data_reader_db.cpp
```

Comment:
Might be worth including the reader functions in the `libweighted_tree` shared library, then this inclusion would not be necessary.  And users of the installed `libweighted_tree` shared library could take advantage of the readers too.

Reply:
After some fiddling around, I think I've got it working. I pushed up some changes and squashed them into 1c30f35, adding **data_reader_db.*** to `libweighted_tree`.